tagstudio/src/core/library.py

```diff
@@ -884,59 +886,62 @@ def refresh_dir(self) -> Generator:
 
         # Scans the directory for files, keeping track of:
         #   - Total file count
-        #   - Files without library entries
-        # for type in TYPES:
-        start_time = time.time()
+        #   - Files without Library entries
+        start_time_total = time.time()
+        start_time_loop = time.time()
+        ext_set = set(self.ext_list)  # Should be slightly faster
         for f in self.library_dir.glob("**/*"):
             try:
-                if (
-                    "$RECYCLE.BIN" not in f.parts
-                    and TS_FOLDER_NAME not in f.parts
-                    and "tagstudio_thumbs" not in f.parts
-                    and not f.is_dir()
+                ext: str = f.suffix.lower()
+                if (ext not in ext_set and self.is_exclude_list) or (
+                    ext in ext_set and not self.is_exclude_list
                 ):
```

Comment:
same comment I left myself in [#494](https://github.com/TagStudioDev/TagStudio/pull/494#issuecomment-2342604899):


> maybe we should add all files into the library (no matter if they are in the ignore list), and only filter them when querying the library. Because the ignore list can eventually change, and it's not intuitive that you have to rescan the directory once again.